hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSNamesystem.java

```diff
@@ -462,13 +462,24 @@ private void appendClientPortToCallerContextIfAbsent() {
               .setSignature(origSignature)
               .build());
     }
+    ctx = CallerContext.getCurrent();
+    if (isFormProxyUser(ctx)) {
+      CallerContext.setCurrent(
+          new CallerContext.Builder(ctx.getContext(), contextFieldSeparator)
+              .append(CallerContext.PROXY_USER_PORT, String.valueOf(Server.getRemotePort()))
+              .setSignature(ctx.getSignature())
+              .build());
+    }
   }
 
   private boolean isClientPortInfoAbsent(CallerContext ctx){
     return ctx == null || ctx.getContext() == null
         || !ctx.getContext().contains(CallerContext.CLIENT_PORT_STR);
   }
 
+  private boolean isFormProxyUser(CallerContext ctx) {
+    return ctx != null && ctx.getContext().contains(CallerContext.REAL_USER_STR);
+  }
```

Comment:
typo: form -> from
Add a null check for getContext as well.
```
  private boolean isFromProxyUser(CallerContext ctx) {
    return ctx != null && ctx.getContext() != null && ctx.getContext()
        .contains(CallerContext.REAL_USER_STR);
  }
```

Reply:
Fixed. Thanks